action.yml

```diff
@@ -0,0 +1,83 @@
+name: Setup Python and Poetry
+description: Setup Python and Poetry. Cribbed from snok/install-poetry.
+inputs:
+  python-version:
+    description: Python version to pass to actions/setup-python@v2.
+    required: false
+    default: "3.x"
+  poetry-version:
+    description: Poetry version to install via pip.
+    required: false
+    default: "1.1.12"
+  debug:
+    description: If set to the string 'true', dump the state of the virtual env after setup.
+    required: false
+    default: "false"
+runs:
+  using: composite
+  steps:
+    - name: Setup Python
+      id: setup-python
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ inputs.python-version }}
+
+    - name: Check for poetry lock
+      run: "test -f poetry.lock || (echo No lock file! && false)"
+      shell: bash
```

Comment:
Is the poetry lock file a necessary requirement? What about for libraries?

Reply:
I'm proposing that we only use this action for projects with a lockfile.

The conventional wisdom for libraries is not to commit a lockfile. That means each CI run would end formally resolving dependencies, which can be sluggish. See for instance [this older attempt in signedjson](https://github.com/matrix-org/python-signedjson/runs/4952543111?check_suite_focus=true): `poetry install --no-interaction` took 22 seconds.

I suspect that building a wheel and installing it in editable mode with pip would be quicker. But more generally I've just kicked the "what do we do about libraries" can down the road.